src/hal/pwm.rs

```diff
@@ -0,0 +1,27 @@
+use crate::pwm::PwmChannel;
+use embedded_hal::PwmPin;
+
+impl PwmPin for PwmChannel {
```

Comment:
Be careful here, `PwmPin` requires the `hal-unproven` feature.

Reply:
`PwmPin` doesn't require the `hal-unproven` feature

Reply:
After re-checking, it seems like only the `Pwm` trait which isn't being implemented here depends on this feature. Forget what I said, my bad.